symengine/ruby/ext/symengine/ruby_basic.c

```diff
@@ -16,15 +16,53 @@ VALUE cbasic_alloc(VALUE klass){
 }
 
 VALUE cbasic_binary_op(VALUE self, VALUE operand2, void (*cwfunc_ptr)(basic_struct*, const basic_struct*, const basic_struct*)){
-    basic_struct *this, *cbasic_operand2, *cresult;
+    basic_struct *this, *cbasic_operand2, *cresult, *num_basic, *den_basic;
     VALUE result;
 
     Data_Get_Struct(self, basic_struct, this);
-    Data_Get_Struct(operand2, basic_struct, cbasic_operand2);
+    VALUE Klass, new_operand2, num, den;
+
+    switch(TYPE(operand2)) {
```

Comment:
You can take out the functionality of this block to a new method like `void sympify (basic new_operand2, VALUE operand2)`